lib/repl.js

```diff
@@ -415,9 +416,11 @@ function REPLServer(prompt,
   });
 
   var sawSIGINT = false;
+  var sawCtrlD = false;
```

Comment:
`self.editorMode` flag tracks the editor mode where as `sawCtrlD` indicates that editor mode is turned off and is not canceled.
`sawCtrlD` flag ensures that vm returned error is non recoverable.